src/vm/vm_memory/memory.rs

```diff
@@ -300,6 +301,29 @@ impl Memory {
     }
 }
 
+impl Display for Memory {
+    fn fmt(&self, f: &mut Formatter) -> std::fmt::Result {
+        writeln!(f, "temp_memory {{")?;
+        for (i, segment) in self.temp_data.iter().enumerate() {
+            for (j, cell) in segment.iter().enumerate() {
+                if let Some(cell) = cell {
+                    writeln!(f, "  ({i},{j}) : {cell}")?;
```

Comment:
Remember that temporary memory segment indexes start from -1, but we store them from 0 (cuz vec). So when printing out the temporary addresses we should be printing `-(i + 1)` instead of `i`